src/Button/Button.test.js

```diff
@@ -29,36 +24,31 @@ describe("Button", () => {
       </Button>
     );
 
-    expect(wrapper.find("button").text()).toEqual(buttonContent);
-    expect(wrapper.find("button").hasClass(className)).toEqual(true);
-    expect(wrapper.find("button").hasClass("button--primary")).toEqual(true);
-    expect(wrapper.find(FontAwesome).hasClass(iconClassName)).toEqual(true);
-    expect(wrapper.find(".content").text()).toEqual(buttonContent);
-
-    wrapper.find("button").simulate("click");
-    expect(onClick).toHaveBeenCalled();
+    const result = queryByText("Button Text");
+    expect(container.firstChild.firstChild.className).toMatch(/fa-download/);
+    expect(result.textContent).toMatch(buttonContent);
   });
 
-  it("should show loading icon instead of regular icon when loading is true", () => {
+  test("onClick method should be called when button is clicked", () => {
     const buttonContent = "Button Text";
+    const className = "a ClassName";
     const icon = "fa-download";
-    const isLoading = true;
+    const iconClassName = "iconClassName";
+    const variant = "primary";
     const onClick = jest.fn();
 
-    const wrapper = mount(
-      <Button icon={icon} isLoading={isLoading} onClick={onClick}>
+    const { container } = render(
+      <Button
+        className={className}
+        icon={icon}
+        variant={variant}
+        onClick={onClick}
+        iconClassName={iconClassName}
+      >
         <div className="content">{buttonContent}</div>
       </Button>
     );
-
-    // loading icon will show
-    expect(wrapper.find(".loadingIcon").exists()).toEqual(true);
-
-    // regular icon will not
-    expect(wrapper.find(`${icon}`).exists()).toEqual(false);
-
-    // can't click while loading
-    wrapper.find("button").simulate("click");
-    expect(onClick).not.toHaveBeenCalled();
+    fireEvent.click(container.firstChild);
```

Comment:
```suggestion
    fireEvent.click(getByRole("button"));
```
Remember the priority list for querying